website/pages/blog/auditing-service-account-keys-with-cloudquery.md

```diff
@@ -0,0 +1,180 @@
+
+As the needs of your services evolve, these Service Accounts tend
+to gain more permissions, which leaves overly permissed keys. All
+services should have their own accounts with the minimum permissions
+required.
+
+# Service Accounts vs Service Account Keys
+
+Service Accounts are accounts created for use by a service.
+These are generally not interactive, often programmatic accounts.
+Service Accounts allow you to associate an identity with a workload,
+and allow permissions to be attached to that identity.
+
+Service Account Keys, are cryptographic keys or shared secrets that
+act as a form a proof to authenticate a call to a Service Account.
+Service Account Keys are just another way to use Service Accounts, albeit
+one with significant risks.
+
+
+Typically, the account keys would look something like this:
+
+GCP Service Account Key:
+```yaml
+id: "key ID"
+private_key: >-
+  -----BEGIN PRIVATE KEY-----
+  PRIVATE_KEY
+  -----END PRIVATE KEY-----
+client_id: "client ID"
+...
+```
+
+AWS Account Key:
+```yaml
+AccessKeyID: "AKIAIOSFODNN7EXAMPLE"
+SecretKey:   "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+```
+
+## Service Account Keys vs Secrets
+
+It is probably tempting to store your Service Account Keys in
+a Key or Secret Management system such as AWS Secret Manager, GCP Secrets Manager,
+Azure KeyVault, or Hashicorp Vault. These systems allow secure
+dissemination of secret material to services or users.
+
+However, this approach is also fraught with peril.
+
+Although secret managers allow secure storage, once the secret material leaves
+the storage system, the security of the secret material is reliant on the client,
+and is vulnerable to leaks.
+Additionally, accessing this secret material requires authentication and authorization,
+implying some preexisting privilege and identity. Regardless of the secure storage,
+the identity problem still persists.
+
+Since you are already using Identity and Access Management to access your secrets, 
+why not lean into the IAM platform and use it for authorizing the end service directly?
+
+## AWS
+
+AWS has a feature called [Access Keys](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html). 
+Access Keys can be disabled by marking them inactive, so at the very least you can quickly pull the authorization without 
+completely getting rid of the keys. 
+
+## GCP
+
+GCP has two types of Service Account Keys: `SYSTEM_MANAGED` and `USER_MANAGED` keys.
+
+`SYSTEM_MANAGED` keys are keys used internally by GCP for creating short-lived service account credentials,
+and for signing blobs and JSON Web Tokens. Users don’t have control over these keys.
+
+`USER_MANAGED` keys are keys created by end users. These service account keys should
+be avoided.
+
+
+# Implementing checks
+
+## Queries
+
+### AWS
+
+Find all Access Keys:
+```sql
+select * from aws_iam_user_access_keys;
+```
+
+Permissions: 
+
+- `iam:CreateAccessKey`
```

Comment:
For read only, `CreateAccessKey`, `DeleteAccessKey`, `UpdateAccessKey` aren't necessary, right?  Unless you want permissions to also manage these access keys?